test/Array/pop4.js

```diff
@@ -0,0 +1,32 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (c) 2022 ChakraCore Project Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+const array = [1, 2, 3];
+
+const testRuns = 100_000;
+let counter = 0;
+
+let proto = [];
+Object.defineProperty(proto, 3, {
+    get: function () {
+        counter++;
+        return 42;
+    }
+});
+array.__proto__ = proto;
+
+function hotFunction() {
+    array.pop();

```

Comment:
We should also test the case where the value is stored - that may work at the moment but worth adding a test for it in case another change later breaks it.